lock/src/main/java/org/infinispan/lock/impl/lock/RequestExpirationScheduler.java

```diff
@@ -0,0 +1,123 @@
+package org.infinispan.lock.impl.lock;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.lock.impl.log.Log;
+
+/**
+ * This class holds the logic to schedule/abort requests that need to be completed at a given time.
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 9.2
+ */
+public class RequestExpirationScheduler {
+
+   private static final Log log = LogFactory.getLog(RequestExpirationScheduler.class, Log.class);
+   private static final boolean trace = log.isTraceEnabled();
+
+   private final ScheduledExecutorService scheduledExecutorService;
+   private final ConcurrentHashMap<String, ScheduledRequest> scheduledRequests = new ConcurrentHashMap<>();
+
+   public RequestExpirationScheduler(ScheduledExecutorService scheduledExecutorService) {
+      this.scheduledExecutorService = scheduledExecutorService;
+   }
+
+   class ScheduledRequest {
+      private String requestId;
```

Comment:
This variable seems superfluous.

Reply:
I'm using it on tests

Reply:
Which is unneeded as well. All the test is doing is verifying that the name is the same after already using the name to retrieve the request. We already retrieved a non null value, we can assume it is tied to the name we used to retrieve it :)

Reply:
ok 🗡